contracts/burner/src/contract.rs

```diff
@@ -24,10 +24,18 @@ pub fn migrate(deps: DepsMut, env: Env, msg: MigrateMsg) -> StdResult<Response>
         to_address: msg.payout.clone(),
         amount: balance,
     };
+
+    let deleted = if msg.delete != 0 {
+        cleanup(deps.storage, Some(msg.delete))
+    } else {
+        0
+    };
```

Comment:
Do we need this case distinction? Shouldn't `cleanup` work perfectly fine with `Some(0)`?

Reply:
Good point. It avoids opening the iterator, but not needed at all.